src/main/java/gregtech/api/util/OverlayedFluidHandler.java

```diff
@@ -0,0 +1,190 @@
+                                if (!(uniqueFluidMap.get(nftfl.getFluidTankList().get()).add(toInsert))) {
+                                    continue;
+                                }
+                        } else {
+                            if (!(uniqueFluidMap.get(mth).add(toInsert))) {
+                                continue;
+                            }
+                        }
+                    }
+                }
+                int spaceInTank = overlayedTank.getCapacity() - overlayedTank.getFluidAmount();
+                int insertable = Math.min(spaceInTank, amountToInsert);
+                if (insertable > 0) {
+                    insertedAmount += insertable;
+                    overlayedTank.setFluidKey(toInsert);
+                    overlayedTank.setFluidAmount(overlayedTank.getFluidAmount() + insertable);
+                    amountToInsert -= insertable;
+                }
+                if (amountToInsert == 0) {
+                    return insertedAmount;
+                }
+            }
+        }
+        // if we still have fluid to insert, insert it into the first tank that can accept it
+        if (amountToInsert > 0) {
+            // loop through the tanks until we find one that can accept the fluid
+            for (int i = 0, tanksLength = this.overlayedTanks.length; i < tanksLength; i++) {
+                OverlayedTank overlayedTank = this.overlayedTanks[i];
+                // if the tank is empty
+                if (overlayedTank.getFluidKey() == null) {
+                    if ((tankDeniesSameFluidFill.contains(overlayed.getTankProperties()[i]) || !this.allowSameFluidFill) ) {
+                        if (overlayed instanceof IMultipleTankHandler) {
+                            IMultipleTankHandler mth = (IMultipleTankHandler) overlayed;
+                            if (mth.getTankAt(i) instanceof NotifiableFluidTankFromList) {
+                                NotifiableFluidTankFromList nftfl = (NotifiableFluidTankFromList) mth.getTankAt(i);
+                                    if (!(uniqueFluidMap.get(nftfl.getFluidTankList().get()).add(toInsert))) {
+                                        continue;
+                                    }
+                            } else {
+                                if (!(uniqueFluidMap.get(mth).add(toInsert))) {
+                                    continue;
+                                }
+                            }
+                        }
+                    }
+                    int canInsertUpTo = Math.min(overlayedTank.getCapacity(), amountToInsert);
+                    if (canInsertUpTo > 0) {
+                        insertedAmount += canInsertUpTo;
+                        overlayedTank.setFluidKey(toInsert);
+                        overlayedTank.setFluidAmount(canInsertUpTo);
+                        amountToInsert -= insertedAmount;
```

Comment:
this is the second part of the loop, where all the tanks are empty. it should insert whatever is lower , the amount when want or the maximum tank capacity

Reply:
yeah, that is being calculated as `canInsertUpTo` on line 129. Since we are inserting the amount of `canInsertUpTo` into the tank, this value should be what is subtracted from `amountToInsert`, not `insertedAmount`, as `insertedAmount` could no longer be accurate to the amount that was inserted by `canInsertUpTo`